src/hooks/useMsgPendingStatuses.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react'
+
+import { msgSubscribe, MsgEvent } from '@/services/msg/msgEvents'
+import { useAppDispatch } from '@/store'
+import { clearPendingMsg, setPendingMsg } from '@/store/pendingMsgsSlice'
+
+const pendingStatuses: Record<MsgEvent, boolean> = {
+  [MsgEvent.CREATE]: true,
+  [MsgEvent.CREATE_FAILED]: false,
+  [MsgEvent.CONFIRM]: true,
+  [MsgEvent.CONFIRM_FAILED]: false,
+  [MsgEvent.CONFIRMATION_SAVED]: false,
+  [MsgEvent.FULLY_CONFIRMED]: false,
+}
+
+const useMsgPendingStatuses = () => {
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    const entries = Object.entries(pendingStatuses) as [keyof typeof pendingStatuses, boolean][]
+
+    const unsubFns = entries.map(([event, isPending]) =>
+      msgSubscribe(event, (detail) => {
+        const messageHash = 'messageHash' in detail && detail.messageHash
+
+        if (!messageHash) {
+          return
+        }
+
+        if (!isPending) {
+          dispatch(clearPendingMsg(messageHash))
+        }
+
+        dispatch(setPendingMsg(messageHash))
```

Comment:
When setting a pending tx we are storing a lot more meta data like `chainId`, `txId`, `status`, `txHash`, `groupKey`, `signerAddress`. Do we need any of them when setting a pending message?

Reply:
The `messageHash` should be unique every time.